pages/api/gas.ts

```diff
@@ -0,0 +1,35 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+  if (req.method !== "GET") {
+    return res.status(400).json({ error: "invalid method" });
+  }
+
+  const gasEndpoint = `https://api.etherscan.io/api?module=gastracker&action=gasoracle&apikey=${process.env.ETHERSCAN_KEY}`;
+  const ethPriceEndpoint = `https://api.etherscan.io/api?module=stats&action=ethprice&apikey=${process.env.ETHERSCAN_KEY}`;
+
+  const gasPrice = await fetch(gasEndpoint, {
```

Comment:
if you want to make this slightly more efficient then make both these fetches at the same time with a `Promise.all([fetch1, fetch2])`